### ISSUE TITLE
USB printing

### ISSUE THREAD

**Issue by joseph-perez** [auctifera-josed/starprnt#14]
Hi there, I was having trouble printing through a usb connection. The usb receipt printer is returned when using the port discovery method but when using the port name to print nothing happens. It looks like maybe it's a problem in the sendCommands method in Communication.m. I noticed the other star receipt cordova plugins suffer from this same problem. Any help is much appreciated, thanks for your time!

Printer being used: TSP100III

**Comment by infoxicator** [comment]
Hi @joseph-perez did you manage to solve the issue?

**Comment by joseph-perez** [comment]
Yeah I think so. So it turns out that everything works correctly it was just the method I was using. When printing through usb you have to use the printRasterReceipt method because some Star receipt printers use line mode a raster mode. Receipt printers seem so complicated! haha

**Comment by infoxicator** [comment]
hi @joseph-perez the printRasterReceipt function doesn't use the command builder interface, so I guess the but is still there. I don't have a USB printer to test, so would you be able to test it for me once I have tracked down the bug?

**Comment by joseph-perez** [comment]
Yeah no worries, I can test it out
